packages/dso-toolkit/gulp/build-svg-spritesheet.js

```diff
@@ -13,135 +13,131 @@ import { sassTransformer } from "./transformers/sass.transformer.js";
-                        selector: "default",
-                      });
+          // Process each transformed SCSS file
+          const parsedStyles = data.map(({ relative, extname, contents }) => {
+            const id = basename(relative, extname); // Extract the file name without the extension
+            const style = contents.toString();
+            const ast = parse(style); // Parse the CSS into an abstract syntax tree (AST)
+
+            // Extract color variants for each CSS rule
+            const variants = ast.stylesheet.rules
+              .filter((rule) => rule.type === "rule") // Keep only CSS rules (ignore comments, etc.)
+              .flatMap((rule) => {
+                const colorDeclaration = rule.declarations.find((d) => d.property === "color");
+                if (!colorDeclaration) return [];
+
+                const color = colorDeclaration.value;
+                return rule.selectors
+                  .map((selector) => {
+                    if (selector === id) {
+                      return { selector: "default", color }; // Mark as default variant
                     }
 
-                    if (
-                      s.startsWith(`${id}:`) &&
-                      !v.some((variantColor) => variantColor.selector === s.substr(`${id}:`.length))
-                    ) {
-                      return current.concat({
-                        color,
-                        selector: s.substr(`${id}:`.length),
-                      });
+                    if (selector.startsWith(`${id}:`)) {
+                      return { selector: selector.slice(id.length + 1), color }; // Extract the variant name
                     }
 
-                    return current;
-                  }, []);
-
-                  v.push(...variantColors);
+                    return null;
+                  })
+                  .filter(Boolean); // Remove null entries
+              });
 
-                  return v;
-                }, []);
+            return { id, variants };
+          });
 
-              return { id, variants };
-            }),
-          );
+          resolve(parsedStyles);
         }),
       );
   });
+}
 
-  return new Promise((resolve, reject) => {
-    gulp
-      .src(`${iconsPath}/*.svg`)
-      .pipe(prettier({ plugins: ["@prettier/plugin-xml"] }))
-      .pipe(
-        svgstore({
-          inlineSvg: true,
-        }),
-      )
-      .pipe(
-        cheerio({
-          run($) {
-            const canvas = 32;
-            const gutter = 10;
-            const symbols = $("symbol").toArray();
-            const positions = symbols.reduce((position, element) => {
-              const symbol = $(element);
-              const id = symbol.attr("id");
-
-              const stylesheet = stylesheets.find((s) => s.id === id);
+async function buildSprite(iconsWithStyles = false) {
+  const stylesheets = await parseStylesheets(); // Parse stylesheets within the function
```

Comment:
De functie `parseStylesheets()` heeft geen afhankelijkheid van `iconsWithStyles`. De return value is voor beide calls van `buildSprite()` en `buildSprite(true)` hetzelfde.

Reply:
Mee eens!